src/extension.ts

```diff
@@ -138,7 +176,47 @@ export function activate(context: vscode.ExtensionContext) {
 
 	}));
 
-	let selected:String = "";
+	context.subscriptions.push(vscode.commands.registerCommand('meld-diff.diffWithClipboard', async () => {
+		const editor = window.activeTextEditor;
+		if (!editor) {
+			window.showErrorMessage("Meld Diff:\nCurrent window is not a file!");
+			return;
+		}
+
+		const clipboardContent = await vscode.env.clipboard.readText();
+		const clipboard = await writeFileOnDisk(clipboardContent);
+
+		//by default compare clipboard against current file
+		let sameContent;
+		let current = editor.document.fileName;
+		const selection = editor.selection;
+		if (!selection.isEmpty) { 
+			//compare against current selection
+			const editorContent = editor.document.getText(selection);
+			current = await writeFileOnDisk(editorContent);
+			sameContent = await areFilesEqual([current, clipboard]);
+		} else if (editor.document.isUntitled) {
+			//compare against untitled file
+			const editorContent = editor.document.getText();
+			current = await writeFileOnDisk(editorContent);
+			sameContent = await areFilesEqual([current, clipboard]);
+		} else if (editor.document.isDirty) {
+			//compore against dirty content but invoke meld with current saved file
```

Comment:
I just saw it now as I developed on top of your changes.
Why do you compare the dirty content but invoke meld with the saved file? Is this also the behavior of the sublime extension?

For me this behavior is confusing.

If I run the command I would expect either to check **and** open the dirty version or check **and** open the saved version of the file.

Background of this question is that I want to add a settings value for the extension were you can select if generally the dirty version or the saved version should be used to compare. So I would make this also equal in the compare to clipboard function.